### ISSUE TITLE
Fix XOR hamming distance calculation

### ISSUE THREAD

**Issue by mchernys** [mashu/LineageCollapse.jl#10]

**Comment by mashu** [comment]
Fixed by using `mismatches` from BioSequences

```julia
"""
    compute_distance(metric::DistanceMetric, x::LongDNA{4}, y::LongDNA{4})::Float32

Compute the distance between two `LongDNA{4}` sequences using the specified distance metric.
"""
function compute_distance(::NormalizedHammingDistance, x::LongDNA{4}, y::LongDNA{4})::Float32
    @assert length(x) == length(y)
    return mismatches(x, y) / length(x)
end
```
For the 
```julia
x = LongDNA{4}(rand("ATGC",30))
y = LongDNA{4}(rand("ATGC",30))
@btime compute_distance_old($x, $y) # gave 91.649ns (6 allocations: 288 bytes)
@btime compute_distance_old($x, $y) # gave 4.125ns (0 allocations: 0 bytes)
```
This is therefore closed by 52b3783d827896e09f076097c90ef7b7a7d11cc8

**Comment by mashu** [comment]
Closed by 52b3783d827896e09f076097c90ef7b7a7d11cc8
